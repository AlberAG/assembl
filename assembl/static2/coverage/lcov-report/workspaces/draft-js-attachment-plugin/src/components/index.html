<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for workspaces/draft-js-attachment-plugin/src/components</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../index.html">All files</a> workspaces/draft-js-attachment-plugin/src/components
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">87.93% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>51/58</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">79.31% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>23/29</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">80% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>12/15</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">87.93% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>51/58</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="AddAttachmentForm.jsx"><a href="AddAttachmentForm.jsx.html">AddAttachmentForm.jsx</a></td>
	<td data-value="16.67" class="pic low"><div class="chart"><div class="cover-fill" style="width: 16%;"></div><div class="cover-empty" style="width:84%;"></div></div></td>
	<td data-value="16.67" class="pct low">16.67%</td>
	<td data-value="6" class="abs low">1/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="3" class="abs low">0/3</td>
	<td data-value="16.67" class="pct low">16.67%</td>
	<td data-value="6" class="abs low">1/6</td>
	</tr>

<tr>
	<td class="file high" data-value="AttachmentButton.jsx"><a href="AttachmentButton.jsx.html">AttachmentButton.jsx</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="14" class="abs high">14/14</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="14" class="abs high">14/14</td>
	</tr>

<tr>
	<td class="file high" data-value="Attachments.jsx"><a href="Attachments.jsx.html">Attachments.jsx</a></td>
	<td data-value="90" class="pic high"><div class="chart"><div class="cover-fill" style="width: 90%;"></div><div class="cover-empty" style="width:10%;"></div></div></td>
	<td data-value="90" class="pct high">90%</td>
	<td data-value="10" class="abs high">9/10</td>
	<td data-value="80" class="pct high">80%</td>
	<td data-value="10" class="abs high">8/10</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="90" class="pct high">90%</td>
	<td data-value="10" class="abs high">9/10</td>
	</tr>

<tr>
	<td class="file high" data-value="DocumentIcon.jsx"><a href="DocumentIcon.jsx.html">DocumentIcon.jsx</a></td>
	<td data-value="90" class="pic high"><div class="chart"><div class="cover-fill" style="width: 90%;"></div><div class="cover-empty" style="width:10%;"></div></div></td>
	<td data-value="90" class="pct high">90%</td>
	<td data-value="10" class="abs high">9/10</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="90" class="pct high">90%</td>
	<td data-value="10" class="abs high">9/10</td>
	</tr>

<tr>
	<td class="file high" data-value="Image.jsx"><a href="Image.jsx.html">Image.jsx</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="18" class="abs high">18/18</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="12" class="abs medium">9/12</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="18" class="abs high">18/18</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Oct 26 2018 15:51:30 GMT+0200 (CEST)
</div>
</div>
<script src="../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../sorter.js"></script>
</body>
</html>
