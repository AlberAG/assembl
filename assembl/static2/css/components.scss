.box {
  background-color: $box-bkg;
  border: $box-border;
  border-radius: $radius;
  box-shadow: $box-shadow;
  padding: 15px 20px 20px;
}

.button-success {
  background-color: $btn-success-bkg;
  border-color: $btn-success-border;
  color: $btn-success-txt;
  font-size: $font-size-s;
  font-weight: 600;
  padding: 8px 50px;

  &:hover {
    background-color: $btn-success-hover-bkg;
    border-color: $btn-success-hover-border;
    color: $btn-success-hover-txt;
  }

  &:focus {
    background-color: $btn-success-hover-bkg;
    border-color: $btn-success-hover-border;
    color: $btn-success-hover-txt;
    outline: none;
  }
}

.button-cancel {
  background-color: $btn-cancel-bkg;
  border-color: $btn-cancel-border;
  color: $btn-cancel-txt;

  &:hover {
    background-color: $btn-cancel-hover-bkg;
    border-color: $btn-cancel-hover-border;
    color: $btn-cancel-hover-txt;
  }
}

.title-1 {
  color: $title1-color;
  font-size: $title1-size;
  font-weight: 700;
  margin: 0 0 15px;
}

.title-2 {
  color: $title2-color;
  font-size: $title2-size;
  font-weight: 700;
  margin: 0 0 10px;
}

.title-3 {
  color: $title3-color;
  font-size: $title3-size;
  font-weight: 700;
  margin: 0 0 5px;
  text-transform: uppercase;
}

.title-4 {
  color: $title4-color;
  font-size: $title4-size;
  font-weight: 700;
  margin: 0 0 10px;
}

.rounded-icon {
  background-color: $grey2;
  border-radius: 10px;
  display: inline-block;
  height: 20px;
  margin: 0 5px 0 0;
  padding: 0 4px;
  width: 20px;

  span {
    color: $white;
    font-size: 12px;
  }
}

.white-icon {
  display: inline-block;
  font-size: 20px;
  line-height: 23px;
  margin: 0 5px 0 0;
  vertical-align: middle;

  span {
    color: $grey2;
  }
}

.black-icon {
  display: inline-block;
  margin: 0 5px 0 0;
}

.profile-icon {
  a {
    &:hover {
      text-decoration: none;
    }
  }

  .username {
    color: $black;
    font-size: $font-size-xs;
    font-weight: 700;
  }

  img {
    border-radius: 11px;
    margin: 0 5px 0 0;
    max-height: 20px;
    width: 20px;
  }
}

.dropdown-navbar {
  font-size: $font-size-xs;
  margin: 0 0 0 15px;
  padding: 4px 0 0;

  .dropdown-toggle {
    color: $black;
    font-size: $font-size-xs;

    &:hover {
      color: $black;
      text-decoration: none;
    }

    &:focus {
      color: $black;
      text-decoration: none;
    }
  }

  .dropdown-menu {
    min-width: 60px;

    a {
      font-size: $font-size-xs;
    }
  }
}

.nav-bar {
  background-color: $navbar-bkg;
  border: $navbar-border;
  height: 80px;

  .navbar-logo {
    padding: 15px 20px;

    img {
      height: 50px;
    }
  }

  .navbar-icons {
    cursor: pointer;
    padding: 28px;

    div {
      margin: 0 10px;
    }
  }

  .navbar-item-menu {
    color: $navbar-text-color;
    font-family: $font-family2;
    font-size: $font-size-xs;
    font-weight: 700;
    line-height: 80px;
    padding: 0 20px;
    text-transform: uppercase;

    &:hover {
      color: $active-link-color;
      text-decoration: none;
    }

    &:focus {
      text-decoration: none;
    }
  }

  .active {
    color: $active-link-color;
  }

  .burger-menu {
    cursor: pointer;
    font-size: 30px;
    padding: 20px
  }
}

.loader {
  text-align: center;

  img {
    margin: 0 40px;
    width: 30px;
  }
}

.app-content {
  margin: 80px 0 0;
}
