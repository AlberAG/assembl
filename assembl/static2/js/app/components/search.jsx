/* eslint no-underscore-dangle: 0, react/no-danger: 0 */
/* global __resourceQuery */
import React from 'react';
import { Localize, Translate, I18n } from 'react-redux-i18n';
import { connect } from 'react-redux';

import {
  ActionBar,
  ActionBarRow,
  BoolMust,
  BoolMustNot,
  CheckboxFilter,
  CheckboxItemList,
  HasChildQuery,
  Hits,
  HitsStats,
  InitialLoader,
  Layout,
  LayoutBody,
  LayoutResults,
  MenuFilter,
  Pagination,
  Panel,
  RangeFilter,
  RangeQuery,
  ResetFilters,
  SearchBox,
  SearchkitManager,
  SearchkitProvider,
  // SelectedFilters,
  SideBar,
  SortingSelector,
  TagFilterConfig,
  TagFilter,
  TermQuery,
  TopBar
} from 'searchkit';
import get from 'lodash/get';
import truncate from 'lodash/truncate';
import Glyphicon from './common/glyphicon';

// Keep the style import here. The reason why it's not in main.scss is because
// we create a searchv1 bundle that includes only the Search component and its
// local styles for v1. There is an additional searchv1.scss file that overrides
// some styles for v1

import '../../../css/views/search.scss';

import DateRangeFilter from './search/DateRangeFilter';
import ProfileLine from './common/profileLine';
import { getConnectedUserId, getDebateId, getLocale } from '../reducers/contextReducer';
import { getPermissionsForConnectedUser } from '../reducers/usersReducer';
import { canUseExpertInterface } from '../utils/permissions';

const FRAGMENT_SIZE = 400;

const truncateText = (text) => {
  return truncate(text,
    { length: FRAGMENT_SIZE, separator: ' ', omission: ' [...]' });
};

const highlightedTextOrTruncatedText = (hit, field) => {
  let text = get(hit, `highlight.${field}`);
  if (text) {
    if (Array.isArray(text)) {
      // take the first highlight fragment
      text = text[0];
    }
    return text;
  }

  text = truncateText(hit._source[field]);
  return text;
};

let Link;
let getUrl;
if (__resourceQuery) { // v1
  // const querystring = require('querystring');
  // const params = querystring.parse(__resourceQuery.slice(1));
  // if (params.v === '1') {
  Link = (props) => {
    return <a href={props.to} dangerouslySetInnerHTML={props.dangerouslySetInnerHTML} />;
  };
  getUrl = (hit) => {
    const slug = document.getElementById('discussion-slug').value;
    const id = hit._source.id;
    switch (hit._type) {
    case 'synthesis':
      return `${slug}/posts/local:Content/${id}`;
    case 'user':
      return undefined;
    case 'idea':
      return `${slug}/idea/local:Idea/${id}`;
    default: // post
      return `${slug}/posts/local:Content/${id}`;
    }
  };
  // }
} else {
  Link = require('react-router').Link;  // eslint-disable-line
  getUrl = (hit) => {
    const id = hit._source.id;
    switch (hit._type) {
    case 'synthesis':
      return `posts/${id}`;
    case 'user':
      return `profile/${id}`;
    case 'idea':
      return `ideas/${id}`;
    default: // post
      return `posts/${id}`;
    }
  };
}

const PublishedInfo = (props) => {
  const { date, userId, userName, className } = props;
  return (
    <div className={className}>
      <Translate value="search.published_the" />{' '}<Localize value={date} dateFormat="date.format" />
      {' '}<Translate value="search.by" />{' '}
      <TagFilter key={userId} field="creator_id" value={userId}>
        <ProfileLine userId={userId} userName={userName} />
      </TagFilter>
    </div>
  );
};

const ImageType = (props) => {
  return (
    <img
      className={props.className}
      src={`/static2/img/icon-${props.type}.svg`}
      role="presentation"
    />
  );
};

const DumbPostHit = (props) => {
  const locale = props.locale;
  const source = props.result._source;
  let subject = get(props.result, `highlight.subject_${locale}`, get(props.result, 'highlight.subject_und'));
  if (!subject) {
    subject = get(source, `subject_${locale}`, source.subject_und);
  }
  let body = get(props.result, `highlight.body_${locale}`, get(props.result, 'highlight.body_und'));
  if (!body) {
    body = get(source, `body_${locale}`, source.body_und);
    body = truncateText(body);
  }
  return (
    <div className={props.bemBlocks.item().mix(props.bemBlocks.container('item'))}>
      <ImageType type={props.result._type} className={props.bemBlocks.item('imgtype')} />
      <div className={props.bemBlocks.item('title')}>
        <Link
          to={getUrl(props.result)}
          dangerouslySetInnerHTML={{ __html: subject }}
        />
      </div>
      <div className={props.bemBlocks.item('content')}>
        <p dangerouslySetInnerHTML={{ __html: body }} />
      </div>
      <PublishedInfo
        className={props.bemBlocks.item('info')}
        date={source.creation_date}
        userId={source.creator_id}
        userName={source.creator_name}
      />
      {/* <div>
        popularity: {source.sentiment_counts.popularity},
        like: {source.sentiment_counts.like},
        disagree: {source.sentiment_counts.disagree}
      </div> */}
    </div>
  );
};

const PostHit = connect(
  (state) => { return { locale: getLocale(state) }; }
)(DumbPostHit);

const SynthesisHit = (props) => {
  const source = props.result._source;
  return (
    <div className={props.bemBlocks.item().mix(props.bemBlocks.container('item'))}>
      <ImageType type={props.result._type} className={props.bemBlocks.item('imgtype')} />
      <div className={props.bemBlocks.item('title')}>
        <Link
          to={getUrl(props.result)}
          dangerouslySetInnerHTML={{ __html: highlightedTextOrTruncatedText(props.result, 'subject') }}
        />
      </div>
      <div className={props.bemBlocks.item('content')}>
        <p dangerouslySetInnerHTML={{ __html: highlightedTextOrTruncatedText(props.result, 'introduction') }} />
        <p dangerouslySetInnerHTML={{ __html: highlightedTextOrTruncatedText(props.result, 'conclusion') }} />
      </div>
      <PublishedInfo
        className={props.bemBlocks.item('info')}
        date={source.creation_date}
        userId={source.creator_id}
        userName={source.creator_name}
      />
    </div>
  );
};


const UserHit = (props) => {
  const source = props.result._source;
  const url = getUrl(props.result);
  const fullname = get(
    props.result, 'highlight.name', props.result._source.name);
  return (
    <div className={props.bemBlocks.item().mix(props.bemBlocks.container('item'))}>
      <ImageType type={props.result._type} className={props.bemBlocks.item('imgtype')} />
      <div className={props.bemBlocks.item('title')}>
        { url ?
          <Link
            to={getUrl(props.result)}
            dangerouslySetInnerHTML={{ __html: fullname }}
          />
        :
          <p dangerouslySetInnerHTML={{ __html: fullname }} />
        }
      </div>
      { source.creation_date ?
        <div className={props.bemBlocks.item('info')}>
          <Translate value="search.member_since" />{' '}<Localize value={source.creation_date} dateFormat="date.format" />
        </div>
      : null }
    </div>
  );
};

const IdeaHit = (props) => {
  const source = props.result._source;
  const shortTitle = highlightedTextOrTruncatedText(props.result, 'short_title');
  const definition = highlightedTextOrTruncatedText(props.result, 'definition');
  const announceTitle = highlightedTextOrTruncatedText(props.result, 'title');
  const announceBody = highlightedTextOrTruncatedText(props.result, 'body');
  return (
    <div className={props.bemBlocks.item().mix(props.bemBlocks.container('item'))}>
      <ImageType type={props.result._type} className={props.bemBlocks.item('imgtype')} />
      <div className={props.bemBlocks.item('title')}>
        <Link
          to={getUrl(props.result)}
          dangerouslySetInnerHTML={{ __html: shortTitle }}
        />
      </div>
      <div className={props.bemBlocks.item('content')}>
        { definition ?
          <div>
            <p dangerouslySetInnerHTML={{ __html: definition }} />
            { get(props.result, 'highlight.definition') && <p><Translate value="search.search_come_from_what_you_need_to_know" /></p> }
          </div>
          : null
        }
        { get(props.result, 'highlight.title') || get(props.result, 'highlight.body') ?
          <div>
            <p dangerouslySetInnerHTML={{ __html: announceTitle }} />
            <p dangerouslySetInnerHTML={{ __html: announceBody }} />
            <p><Translate value="search.search_come_from_announcement" /></p>
          </div>
          : null
        }
      </div>
      <div className={props.bemBlocks.item('info')}>
        {source.num_posts}
        <span className={props.bemBlocks.item('icon-message')}>
          <Glyphicon glyph="message" color="purple" size={20} desc="Number of contributions" />
        </span>
        {source.num_contributors}
        <span className={props.bemBlocks.item('icon-avatar')}>
          <Glyphicon glyph="avatar" color="purple" size={20} desc="Number of users" />
        </span>
      </div>
    </div>
  );
};

const HitItem = (props) => {
  switch (props.result._type) {
  case 'synthesis':
    return <SynthesisHit {...props} />;
  case 'user':
    return <UserHit {...props} />;
  case 'idea':
    return <IdeaHit {...props} />;
  default: // post
    return <PostHit {...props} />;
  }
};

const NoPanel = (props) => {
  return (
    <div>{props.children}</div>
  );
};


const queryFields = [
  'title',  // idea announcement
  'body',  // idea announcement
  'short_title',  // idea
  'definition',  // idea
  'name',  // user
  'subject',  // synthesis
  'introduction',  // synthesis
  'conclusion',  // synthesis
  'subject_und', // post
  'subject_fr', // post
  'subject_en', // post
  'body_und', // post
  'body_fr', // post
  'body_en' // post
];

// default is fragment_size 100 and number_of_fragments 5,see
// https://www.elastic.co/guide/en/elasticsearch/reference/current/search-request-highlighting.html#_highlighted_fragments
const customHighlight = {
  fields: {
    body: { fragment_size: FRAGMENT_SIZE, number_of_fragments: 1 }
  }
};
// setting it for body field seems to apply the setting to all queryFields

export class SearchComponent extends React.Component {

  constructor(props) {
    super(props);
    const host = '/';
    this.searchkit = new SearchkitManager(host, { searchOnLoad: false });
    this.searchkit.setQueryProcessor((plainQueryObject) => {
      // rewrite the query to filter on the current discussion
      const modifiedQuery = plainQueryObject;
      const filters = [];
      // we need to rewrite the query as a bool query to be compatible ES 5
      if (modifiedQuery.filter) {
        // modifiedQuery.filter is always an Object,
        // it's a range filter for example
        // or a bool must when there is more than 1 filter
        filters.push(modifiedQuery.filter);
        delete modifiedQuery.filter;
      }
      filters.push({ term: { discussion_id: props.discussionId } });
      let simpleQueryString;
      if (modifiedQuery.query) {
        simpleQueryString = modifiedQuery.query.simple_query_string;
        delete modifiedQuery.query.simple_query_string;
      }
      modifiedQuery.query = { bool: { filter: filters } };
      if (simpleQueryString) {
        this.setState({ show: true, queryString: simpleQueryString.query });
        simpleQueryString.query = `${simpleQueryString.query}*`;
        modifiedQuery.query.bool.must = [
          { simple_query_string: simpleQueryString }
        ];
      } else {
        this.setState({ queryString: null });
      }
      return modifiedQuery;
    });
    const translate = I18n.t.bind(I18n);
    this.searchkit.translateFunction = (key) => {
      return translate(`search.${key}`);
    };
    this.state = { show: false, queryString: null };
  }

  render() {
    const { isExpert, connectedUserId, discussionId } = this.props;
    let messagesSelected = false;
    let usersSelected = false;
    if (this.searchkit.state) {
      messagesSelected = this.searchkit.state.type.indexOf('post') >= 0;
      usersSelected = this.searchkit.state.type.indexOf('user') >= 0;
    }
    const sorts = [
      { label: 'By relevance', field: '_score', order: 'desc', defaultOption: true },
      { label: 'Most recent first', field: 'creation_date', order: 'desc' },
      { label: 'Oldest first', field: 'creation_date', order: 'asc' }
    ];
    if (messagesSelected) {
      sorts.push(
        { label: 'Most popular messages',
          key: 'popularity_desc',
          fields: [
            { field: 'sentiment_counts.popularity', options: { order: 'desc' } },
            { field: 'creation_date', options: { order: 'desc' } }
          ]
        }
      );
      sorts.push(
        { label: 'Less popular messages',
          key: 'popularity_asc',
          fields: [
            { field: 'sentiment_counts.popularity', options: { order: 'asc' } },
            { field: 'creation_date', options: { order: 'desc' } }
          ]
        }
      );
    }
    return (
      <SearchkitProvider searchkit={this.searchkit}>
        <Layout size="l">
          <TopBar>
            <SearchBox
              autofocus={false}
              searchOnChange
              searchThrottleTime={500}
              queryFields={queryFields}
            />
            <button
              className="btn btn-default btn-sm" id="search-expand"
              onClick={() => {
                this.setState({ show: !this.state.show }, () => {
                  if (this.state.show && !this.searchkit.hasHits()) {
                    this.searchkit.reloadSearch();
                  }
                });
              }}
            >
              {this.state.show ?
                <Translate value="search.collapse_search" />
              :
                <Translate value="search.expand_search" />
              }
            </button>
          </TopBar>
          <LayoutBody className={!this.state.show ? 'hidden' : null}>
            <SideBar>
              <ResetFilters />
              {/* <SelectedFilters /> */}
              <MenuFilter
                listComponent={CheckboxItemList}
                field="_type"
                id="type"
                title={I18n.t('search.Categories')}
              />
              <Panel title={I18n.t('search.Messages')} className={messagesSelected ? null : 'hidden'}>
                <MenuFilter
                  containerComponent={NoPanel}
                  listComponent={CheckboxItemList}
                  field="sentiment_tags"
                  id="sentiment_tags"
                  title={I18n.t('search.Messages')}
                />
                { connectedUserId ?
                  <div className="sk-panel">
                    <CheckboxFilter
                      containerComponent={NoPanel}
                      id="mymessages"
                      title={I18n.t('search.My messages')}
                      label={I18n.t('search.My messages')}
                      filter={TermQuery('creator_id', connectedUserId)}
                    />
                    <CheckboxFilter
                      containerComponent={NoPanel}
                      id="messages-in-response"
                      title={I18n.t('search.Messages in response to my contributions')}
                      label={I18n.t('search.Messages in response to my contributions')}
                      filter={TermQuery('parent_creator_id', connectedUserId)}
                    />
                  </div> : null }
              </Panel>
              { isExpert ?
                <Panel title={I18n.t('search.Participants')} className={usersSelected ? null : 'hidden'}>
                  <CheckboxFilter
                    containerComponent={NoPanel}
                    id="creative-participants"
                    title={I18n.t('search.Creative participants')}
                    label={I18n.t('search.Creative participants')}
                    filter={
                      HasChildQuery('post', BoolMust([
                        TermQuery('discussion_id', discussionId),
                        TermQuery('parent_id', 0)
                      ]))
                    }
                  />
                  <CheckboxFilter
                    containerComponent={NoPanel}
                    id="reactive-participants"
                    title={I18n.t('search.Reactive participants')}
                    label={I18n.t('search.Reactive participants')}
                    filter={
                      BoolMust([
                        HasChildQuery('post', BoolMust([
                          TermQuery('discussion_id', discussionId),
                          RangeQuery('parent_id', { gt: 0 })
                        ])),
                        BoolMustNot(
                          HasChildQuery('post', BoolMust([
                            TermQuery('discussion_id', discussionId),
                            TermQuery('parent_id', 0)
                          ]))
                        )
                      ])
                    }
                  />
                  <CheckboxFilter
                    containerComponent={NoPanel}
                    id="learning-participants"
                    title={I18n.t('search.Learning participants')}
                    label={I18n.t('search.Learning participants')}
                    filter={
                      BoolMustNot([
                        HasChildQuery('post',
                          TermQuery('discussion_id', discussionId)
                        )
                      ])
                    }
                  />
                  <CheckboxFilter
                    containerComponent={NoPanel}
                    id="participants-peers"
                    title={I18n.t('search.Participants pleased by their peers')}
                    label={I18n.t('search.Participants pleased by their peers')}
                    filter={
                      HasChildQuery('post', BoolMust([
                        TermQuery('discussion_id', discussionId),
                        RangeQuery('sentiment_counts.like', { gt: 0 })
                      ]))
                    }
                  />
                </Panel>
              : null }
              <TagFilterConfig id="creator_id" title="Participant" field="creator_id" />
              <Panel title={I18n.t('search.Sort')}>
                <SortingSelector
                  options={sorts}
                  listComponent={CheckboxItemList}
                />
              </Panel>
              <RangeFilter
                field="creation_date"
                id="creation_date"
                title={I18n.t('search.Filter by date')}
                rangeComponent={DateRangeFilter}
                min={946684800000}
                max={new Date().getTime()}
              />
            </SideBar>
            <LayoutResults>
              <ActionBar>
                <ActionBarRow>
                  <HitsStats />
                </ActionBarRow>
              </ActionBar>
              <Hits
                hitsPerPage={5}
                highlightFields={queryFields}
                customHighlight={customHighlight}
                itemComponent={HitItem}
                mod="sk-hits-list"
              />
              <InitialLoader />
              <Pagination showNumbers />
            </LayoutResults>
          </LayoutBody>
        </Layout>
      </SearchkitProvider>
    );
  }

}

const mapStateToProps = (state) => {
  const permissions = getPermissionsForConnectedUser(state);
  return {
    isExpert: canUseExpertInterface(permissions),
    connectedUserId: getConnectedUserId(state),
    discussionId: getDebateId(state)
  };
};

const ConnectedSearch = connect(mapStateToProps)(SearchComponent);
export default ConnectedSearch;