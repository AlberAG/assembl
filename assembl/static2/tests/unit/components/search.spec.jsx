import React from 'react';
import renderer from 'react-test-renderer';
import { SearchkitManager, SearchkitProvider } from 'searchkit';
import { Router } from 'react-router';

import * as search from '../../../js/app/components/search';

jest.mock('react-router');
jest.mock('../../../js/app/router');

const fakeBemBlocks = {
  container: name => name,
  item: name => ({
    mix: otherName => (name ? `div__${name}__${otherName}` : `div__${otherName}`)
  })
};

describe('Hits components', () => {
  let context;
  let searchkit;

  beforeEach(() => {
    context = {
      router: new Router()
    };
    searchkit = SearchkitManager.mock();
  });

  afterEach(() => {
    context = null;
    searchkit = null;
  });

  describe('PostHit component', () => {
    const { PostHit } = search;

    it('should render a post hit', () => {
      const props = {
        bemBlocks: fakeBemBlocks,
        locale: 'en',
        result: {
          _type: 'post',
          _source: {
            creation_date: '2018-08-08',
            creator_id: 'foo',
            creator_name: 'Paxton Pouros',
            subject_en: 'We need to quantify the optical HTTP panel!',
            body_en: 'We need to connect the multi-byte SQL interface!',
            idea_id: 'fakeIdeaId',
            post_id: 'fakePostId',
            sentiment_counts: {
              like: 2,
              disagree: 3,
              dont_understand: 3,
              more_info: 1
            }
          }
        }
      };

      const component = renderer.create(
        <SearchkitProvider searchkit={searchkit}>
          <PostHit {...props} />
        </SearchkitProvider>,
        {
          context: context
        }
      );
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('DumbExtractHit component', () => {
    const { DumbExtractHit } = search;
    it('should render an extract hit', () => {
      const props = {
        bemBlocks: fakeBemBlocks,
        locale: 'en',
        result: {
          _type: 'extract',
          _source: {
            creation_date: '2018-08-08',
            creator_id: 'foo',
            creator_name: 'Jewell Pouros',
            subject_en: 'We need to quantify the optical HTTP panel!',
            body: 'We need to connect the multi-byte SQL interface!'
          }
        }
      };

      const component = renderer.create(
        <SearchkitProvider searchkit={searchkit}>
          <DumbExtractHit {...props} />
        </SearchkitProvider>,
        {
          context: context
        }
      );
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});