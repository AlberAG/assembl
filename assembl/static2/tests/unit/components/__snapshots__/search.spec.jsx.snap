// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hits components DumbExtractHit component should render an extract hit 1`] = `
<div
  className="div__item"
>
  <span
    className="[object Object] assembl-icon-discussion"
  />
  <div
    className={
      Object {
        "mix": [Function],
      }
    }
  />
  <div
    className={
      Object {
        "mix": [Function],
      }
    }
  >
    <p
      dangerouslySetInnerHTML={
        Object {
          "__html": "We need to connect the multi-byte SQL interface!",
        }
      }
    />
  </div>
  <div
    className={
      Object {
        "mix": [Function],
      }
    }
  >
    <span
      className={undefined}
      style={undefined}
    >
      Harvested on
    </span>
     
    <span
      className={undefined}
      style={undefined}
    >
      August 8th, 2018
    </span>
     
    <span
      className={undefined}
      style={undefined}
    >
      by
    </span>
     
    <div
      className="sk-tag-filter"
      onClick={[Function]}
    >
      <div
        className="profileLine"
      >
        <div
          className="inline"
        >
          <img
            alt="Jewell Pouros"
            height="30"
            onError={[Function]}
            src="/user/id/foo/avatar/30"
            width="30"
          />
        </div>
        <div
          className="user"
        >
          <div
            className="creator"
          >
            Jewell Pouros
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Hits components PostHit component should render a post hit 1`] = `
<div
  className="div__item"
>
  <span
    className="[object Object] assembl-icon-discussion"
  />
  <div
    className={
      Object {
        "mix": [Function],
      }
    }
  />
  <div
    className={
      Object {
        "mix": [Function],
      }
    }
  >
    <p
      dangerouslySetInnerHTML={
        Object {
          "__html": "We need to connect the multi-byte SQL interface!",
        }
      }
    />
    <div>
      <div
        className="emoticon LikeSentimentOfPost"
        title="Like"
      />
      <div
        className="emoticonValue"
      >
        2
      </div>
      <div
        className="emoticon DisagreeSentimentOfPost"
        title="Disagree"
      />
      <div
        className="emoticonValue"
      >
        3
      </div>
      <div
        className="emoticon DontUnderstandSentimentOfPost"
        title="Did not get it"
      />
      <div
        className="emoticonValue"
      >
        3
      </div>
      <div
        className="emoticon MoreInfoSentimentOfPost"
        title="More info please?"
      />
      <div
        className="emoticonValue"
      >
        1
      </div>
    </div>
  </div>
  <div
    className={
      Object {
        "mix": [Function],
      }
    }
  >
    <span
      className={undefined}
      style={undefined}
    >
      Published on
    </span>
     
    <span
      className={undefined}
      style={undefined}
    >
      August 8th, 2018
    </span>
     
    <span
      className={undefined}
      style={undefined}
    >
      by
    </span>
     
    <div
      className="sk-tag-filter"
      onClick={[Function]}
    >
      <div
        className="profileLine"
      >
        <div
          className="inline"
        >
          <img
            alt="Paxton Pouros"
            height="30"
            onError={[Function]}
            src="/user/id/foo/avatar/30"
            width="30"
          />
        </div>
        <div
          className="user"
        >
          <div
            className="creator"
          >
            Paxton Pouros
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
